homeassistant/components/mysensors.py

```diff
@@ -7,59 +7,74 @@
 import logging
 import socket
 
+import voluptuous as vol
+
 from homeassistant.bootstrap import setup_component
+import homeassistant.helpers.config_validation as cv
 from homeassistant.const import (ATTR_BATTERY_LEVEL, CONF_OPTIMISTIC,
                                  EVENT_HOMEASSISTANT_START,
                                  EVENT_HOMEASSISTANT_STOP, STATE_OFF, STATE_ON)
-from homeassistant.helpers import discovery, validate_config
+from homeassistant.helpers import discovery
 from homeassistant.loader import get_component
 
-CONF_GATEWAYS = 'gateways'
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_NODE_ID = 'node_id'
+ATTR_CHILD_ID = 'child_id'
+ATTR_DESCRIPTION = 'description'
+ATTR_DEVICE = 'device'
+CONF_BAUD_RATE = 'baud_rate'
 CONF_DEVICE = 'device'
 CONF_DEBUG = 'debug'
+CONF_GATEWAYS = 'gateways'
 CONF_PERSISTENCE = 'persistence'
 CONF_PERSISTENCE_FILE = 'persistence_file'
-CONF_VERSION = 'version'
-CONF_BAUD_RATE = 'baud_rate'
 CONF_TCP_PORT = 'tcp_port'
 CONF_TOPIC_IN_PREFIX = 'topic_in_prefix'
 CONF_TOPIC_OUT_PREFIX = 'topic_out_prefix'
 CONF_RETAIN = 'retain'
-DEFAULT_VERSION = '1.4'
+CONF_VERSION = 'version'
+DEFAULT_VERSION = 1.4
 DEFAULT_BAUD_RATE = 115200
 DEFAULT_TCP_PORT = 5003
-
 DOMAIN = 'mysensors'
-DEPENDENCIES = []
+GATEWAYS = None
 MQTT_COMPONENT = 'mqtt'
 REQUIREMENTS = [
     'https://github.com/theolind/pymysensors/archive/'
     '8ce98b7fb56f7921a808eb66845ce8b2c455c81e.zip#pymysensors==0.7.1']
-_LOGGER = logging.getLogger(__name__)
-ATTR_NODE_ID = 'node_id'
-ATTR_CHILD_ID = 'child_id'
-ATTR_DEVICE = 'device'
 
-GATEWAYS = None
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_GATEWAYS): vol.All(cv.ensure_list, [
+            {
+                vol.Required(CONF_DEVICE): cv.string,
+                vol.Optional(CONF_PERSISTENCE_FILE): cv.string,
```

Comment:
hrm... well we could move persist to the same level as the persistence file - that would allow you to persist some devices, and not others - more granularity